packages/utils/node-resolver-rs/src/fs.rs

```diff
@@ -1,12 +1,14 @@
 use std::{
+  cell::RefCell,
   io::Result,
   path::{Path, PathBuf},
 };
 
 use crate::path::canonicalize;
 use dashmap::DashMap;
+use pnp::fs::{LruZipCache, VPath, ZipCache};
 
-pub trait FileSystem: Send + Sync {
```

Comment:
I don't exactly know what this entails ... without it, the RefCell didn't work. I needed the RefCell to bypass the lack of `mut` qualifier on `&self`, so that the `zip_cache` could be updated. Is there a better approach?

Reply:
`RefCell` only allows single threaded access, but `Send + Sync` requires multi threaded access. You can use a `Mutex` or `RwLock` to allow interior mutability similar to `RefCell` but with support for multiple threads.